### DISCUSSION TITLE
ListView multiple selected items binding

### DISCUSSION THREAD

**Discussion by iot-crazy** [unoplatform/uno#8360] [Q&A]
Hi All, 
I'm using a ListView in multiple selection mode, and I'm trying to find a way to bind the multiple selected items to a collection. But, all the "Selected-" properties support only single selection.

So far, I'm trying to get around this by adding a 'Selected' property on the item model, setting this with a setter as seen here [https://stackoverflow.com/questions/14714723/listview-isselected](https://stackoverflow.com/questions/14714723/listview-isselected)

However, this means I need to observe all the items in the collection and write code to update another collection with the selected items, and this needs to be bidirectional. 

Maybe a real life example would help?

Imagine 2 List Views, one is a list of baskets, the other is a list of products.
Each basket can contain zero or many products.  
A ListView shows all the basket and an icon for each product that is in that basket.  When a user clicks on a basket in that first list, the 2nd list needs to show those products selected. If the user then changes the selection of products, the collection in the first ListView needs to update the items in the basket.

As you can see, each basket in the collection of baskets, contains a list of products, so would be nice if selecting a basket, I could just bind the collection to the 2nd ListView.

**Comment by MartinZikmund** [comment]
I would use the [`SelectedItems`](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.listviewbase.selecteditems?view=winrt-22000) property in conjunction with [`SelectionChanged` event](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.primitives.selector.selectionchanged?view=winrt-22000). The event should give you information on which items were added and which removed each time. Data-binding in this case is not too easy, as the `SelectedItems` is just a normal property.

**Reply by iot-crazy** [reply]
Thank you, I'm looking into this today.  Your suggestion will give me a direction from the control to the outside collection so I'll start there and also look at how to make it bidirectional.

**Reply by Xiaoy312** [reply]
You can take some inspiration from this control: https://github.com/unoplatform/uno.toolkit.ui/blob/main/src/Uno.Toolkit.UI/Controls/Chips/ChipGroup.cs
It support bidirectional bindings. While it is implemented on the control level, but it shouldn't be hard to translate the implementation onto an attached property.

**Reply by iot-crazy** [reply]
Thank you, I'll take a look. In the meantime, I've got it working, although it's not very pretty code, it does prove the concept.  Any feedback very much appreciated!

[https://github.com/iot-crazy/UnoListViewMultiSelect](https://github.com/iot-crazy/UnoListViewMultiSelect)

**Reply by iot-crazy** [reply]
There is an issue with this that if I try to build for any other startup project than UWP, the build fails. In the editor, it says that lstproducts does not exist in ProductListView.xaml.cs 
Also, I get this in the generated file for the mainpage

Setters = 
							{
								new global::Windows.UI.Xaml.Setter(global::Windows.UI.Xaml.Controls.ListViewItem.IsSelectedProperty, () => (bool)
								#Error // Binding could not be found.
								)
								,
							}

Two errors here
1. Invalid expression term ')'
2. Preprocessor directive expected

**Reply by iot-crazy** [reply]
I finally got this building and deployed to Android, but now i find that turning the screen breaks the app and items no longer selected or deselect, but sometimes they do. If I turn back to portrait, then it starts working again if I click on another item first.

**Comment by TonyK-FmGroup** [comment]
Hello All,

My final solution on this was to use an additional user control inside the basket, which listed the products. This way, there was no need for event listeners, instead I just had to bind an `ObservableCollection<Products>` to the control's `DependencyProperty` 
If this is unclear, please reply and I'll get a demo up on a repo
